Add documentation for supporting resending account verification email (#5355)
wso2/docs-is: en/includes/guides/account-configurations/user-onboarding/self-registration.md

> Once you have identified the `emailVerified` status of the user and the method by which the user is onboarded (self-registration or onboarded by an administrator), you can enforce any access restrictions for that user through your application logic.

Reviewer: ```suggestion
Once you have identified the `emailVerified` status of the user and the method used to onboard the user (either through self-registration or by an administrator), you can enforce any access restrictions through your application logic.
```

Author: This content is not something introduced with my improvment. However, will update this as well .

Author: Resolved via https://github.com/wso2/docs-is/pull/5355/commits/8aab8b464237b6ab3565afe5ecc80b52e7d3197a

{% if product_name == "Asgardeo" or (product_name == "WSO2 Identity Server" and is_version > "7.1.0" ) %}
## Resend account verification email

> Resending the account verification email helps users who didn't receive the initial email, deleted it, or found the verification link expired. This ensures users can complete the verification process and gain appropriate access to the application.

Author: Agree with you.  But for completeness thought of adding it . Will remove.

> Resending the account verification email helps users who didn't receive the initial email, deleted it, or found the verification link expired. This ensures users can complete the verification process and gain appropriate access to the application.

Author: Resolved via https://github.com/wso2/docs-is/pull/5355/commits/8aab8b464237b6ab3565afe5ecc80b52e7d3197a

!!! tip

    To implement this scenario, you must enforce account verification for self-registered users and activate accounts immediately without waiting for verification.</br></br> Learn more in [enable/disable self-registration](#enabledisable-self-registration).

> - To resend the account verification email for your own account, use the resend-code API as shown below.

Reviewer: shall we link to the resend code API doc?

Reviewer: ```suggestion
- To resend the account verification email for the authenticated user, use the resend-code API as shown below.
```

Author: > shall we link to the resend code API doc?

In asgardeo we don't have exposed this API

Author: Resolved via https://github.com/wso2/docs-is/pull/5355/commits/8aab8b464237b6ab3565afe5ecc80b52e7d3197a

{% if product_name == "Asgardeo"%}
!!! abstract ""

    === "Request format"

        ```curl
        curl -X 'POST' \
        'https://api.asgardeo.io/t/{organization_name}/api/identity/user/v1.0/me/resend-code' \
        -H 'Authorization: Bearer <access_token>' \
        -H 'Content-Type: application/json' \
        -d '{
            "properties": [
                {
                    "key": "RecoveryScenario",
                    "value": "SELF_SIGN_UP"
                }
            ]
        }'
        ```
    === "Sample request"

        ```curl
        curl -X 'POST' \
        'https://api.asgardeo.io/t/{organization_name}/api/identity/user/v1.0/me/resend-code' \
        -H 'Authorization: Bearer <access_token>' \
        -H 'Content-Type: application/json' \
        -d '{
            "properties": [
                {
                    "key": "RecoveryScenario",
                    "value": "SELF_SIGN_UP"
                }
            ]
        }'
        ```

    ---
    **Response**
    ```
    HTTP/1.1 201 Created
    ```

> {% else %}

Reviewer: Hope this is applicable for all IS versions. If not let's improve the conditional to display this only for relevant IS versions

Author: This is only applicable for IS versions above 7.1. The condition is already applied.

Author: Resolved via https://github.com/wso2/docs-is/pull/5355/commits/8aab8b464237b6ab3565afe5ecc80b52e7d3197a
